enhancements/network/dpu-ovnkube-operator.md

```diff
@@ -0,0 +1,186 @@
+  - manages the ovn-kube components that serve the tenant cluster.
+  - enable the swichedev mode on BlueField card.
+- Define the ovn-kube upgrade mechanism in such a deployment.
+
+### Non-Goals
+
+Support for network plugins other than OVN Kubernetes.
+This enhancement does not address the two-cluster installation.
+
+## Pre-requisites
+
+- This operator needs to access the apiserver of the tenant cluster, we need to
+make sure that the pods network in infra cluster can connect to the apiserver of
+the tenant cluster in some way.
+- One custom MachineConfigPool will be created for the BlueField nodes on which
+hardware offloading needs to be enabled.
+
+## Proposal
+
+We will create a namespaced CRD OVNKubeConfig.smartnic.openshift.io for the
+DPU OVNKube Operator. One custom resource of this CRD shall be created for
+the tenant cluster. The operator will take this custom resource as input to
+render and create the objects of ovnkube smart-nic components. The operator will
+also create a MachineConfig CR which injects a systemd service to enable
+switchdev mode on BF-2 node.
+
+### User Stories
+
+As a cluster admin I want to manage the ovn-kubernetes components in the infra
+cluster, and these components can be controlled by the ovn-kubernetes control plane in the tenant cluster.
+
+### Risks and Mitigations
+
+## Design Details
+
+The DPU OVNKube Operator will fetch the necessary information from the
+tenant cluster, then create the following objects in the infra cluster:
+- The ovnkube-node DaemonSet running in the SmartNIC mode
+- The ServiceAccount and corresponding RBAC definition
+- The ConfigMap that contains ​​ovnkube-config of the tenant cluster
+- The ComfgMap and Secret that contains the CA and signed certificate of the
+- tenant cluster OVN
+- The ConfigMap CA and signed certificate of the tenant cluster API server
+
+The DPU OVNKube Operator will be responsible for enable the switchdev mode on
+BF2 cards. We assume all the BF2 cards will be put into a custom
+MachineConfigPool,
+
+Currently, we are only targeting the 1to1 deployment mode, meaning that
+there is 1to1 mapping between the infra cluster and the tenant cluster. However,
+it is possible to have one infra cluster to serve more than one tenant cluster
+in the future. We may want to create one custom resource in a different
+namespace for each tenant cluster.
+
+### API
+Example configurations:
+
+```yaml
+apiVersion: smartnic.openshift.io/v1
+kind: OVNKubeConfig
+metadata:
+  name: cluster
+  namespace: tenant-cluster-1
+spec:
+  kubeConfigFile: tenant-cluster-1-kubeconf
+  poolName: bf2-worker
+```
+
+1. `kubeConfigFile` stores the secret name of the tenant cluster kubeconfig
+   file. The operator uses this to access the api-server of the tenant
+   cluster.
+2. `poolName` specify the name of the MachineConfigPool CR which contains all
+   the BF2 nodes. The operator inherits the `spec.nodeSelector` of the MCP to
+   render the ovnkube-node daemonset.
```

Comment:
"inherits" implies some hierarchical relationship. I think you just mean that the operator "uses" or "copies" the `spec.nodeSelector` of the MCP?